lib/dataContract/DataContract.js

```diff
@@ -145,6 +147,29 @@ class DataContract {
     return this.definitions;
   }
 
+  /**
+   * Get properties with `contentEncoding` constraint
+   *
+   * @param {string} type
+   *
+   * @return {Object}
+   */
+  getEncodedProperties(type) {
+    if (!this.isDocumentDefined(type)) {
+      throw new InvalidDocumentTypeError(type, this);
+    }
+
+    if (this.encodedProperties) {
```

Comment:
New document type will overwrite previous type cache. Let's add type keys to the `encodedProperties` object.